components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Runtime/Extensions/OpcConfigEx.cs

```diff
@@ -40,38 +41,58 @@ public static class OpcConfigEx {
 
             applicationConfiguration.CertificateValidator.CertificateValidation += handler;
 
-            X509Certificate2 certificate = null;
+            var configuredSubject = applicationConfiguration.SecurityConfiguration
+                .ApplicationCertificate.SubjectName;
+            applicationConfiguration.SecurityConfiguration.ApplicationCertificate.SubjectName = 
+                applicationConfiguration.ApplicationName;
+            applicationConfiguration.CertificateValidator
+                .Update(applicationConfiguration.SecurityConfiguration).ConfigureAwait(false);
```

Comment:
is this intentionally running async with await/wait?

Reply:
not sure what was the intention here, I'll change it to async method though